components/automate-ui/src/app/pages/project/confirm-apply-stop-modal/confirm-apply-stop-modal.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, Input, OnChanges, Output, EventEmitter } from '@angular/core';
+import { ApplyRulesStatus } from 'app/entities/projects/project.reducer';
+import * as moment from 'moment';
+
+@Component({
+  selector: 'app-confirm-apply-stop-modal',
+  templateUrl: './confirm-apply-stop-modal.component.html',
+  styleUrls: [ './confirm-apply-stop-modal.component.scss' ]
+})
+export class ConfirmApplyStopModalComponent implements OnChanges {
+
+  @Input() applyRulesStatus: ApplyRulesStatus;
+
+  @Output() confirm: EventEmitter<void> = new EventEmitter();
+
+  @Output() cancel: EventEmitter<void> = new EventEmitter();
+
+  progressValue: number;
+
+  progressPrefixText: string;
+
+  progressSuffixText: string;
+
+  ngOnChanges() {
```

Comment:
Seems like there is enough going on here to merit unit testing these calculations, if it is not too cumbersome to set up.